File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/UserRegistrationStatusDto.java
package de.sovity.authorityportal.api.model;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "Possible user registration statuses.", enumAsRef = true)
public enum UserRegistrationStatusDto {
    FIRST_USER,         // Special status indicating the first user, only used on the first user registration
    INVITED,            // User that has been invited by another participant
    CREATED,            // Edge-case status, used between a self-registration and the first login
    ONBOARDING,         // User that has self-registered but needs to fill in missing data on login
    PENDING,            // User that has self-registered and is waiting for approval
    ACTIVE,             // User that has been approved and is active
    REJECTED,           // User that has been rejected
    DEACTIVATED         // User that has been deactivated, no access to the portal
}

File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/UserRegistrationStatusDto.kt
/*
 * Copyright (c) 2024 sovity GmbH
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contributors:
 *      sovity GmbH - initial implementation
 */

package de.sovity.authorityportal.api.model

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Possible user registration statuses.", enumAsRef = true)
enum class UserRegistrationStatusDto {
    INVITED,  // User that has been invited to be a participant user
    ONBOARDING,  // User that has been invited to be a participant admin
    PENDING,  // User awaiting approval of his self-registration
    ACTIVE,  // Active fully registered approved User
    REJECTED,  // User who got his self-registration rejected
    DEACTIVATED // User that has been deactivated, no access to the portal
}

